### ISSUE TITLE
sveltekit post formdata fails when using bunx instead of node

### ISSUE THREAD

**Issue by codegod100** [oven-sh/bun#4234]
### What version of Bun is running?

0.7.3

### What platform is your computer?

gitpod Ubuntu 22.04.3 LTS \n \l

### What steps can reproduce the bug?

```
  const body = new FormData();
  body.append("client_name", "tootbooster");
  body.append("redirect_uris", `${REDIRECT_URI}/${host}`);
  body.append("scopes", "read write push");
  const url = `https://${host}/api/v1/apps`;
  const res = await fetch(url, {
    method: "POST",
    body,
  });
```

### What is the expected behavior?

```
trying
{
  res: Response {
    [Symbol(realm)]: null,
    [Symbol(state)]: {
      aborted: false,
      rangeRequested: false,
      timingAllowPassed: true,
      requestIncludesCredentials: true,
      type: 'default',
      status: 200,
      timingInfo: [Object],
      cacheState: '',
      statusText: 'OK',
      headersList: [HeadersList],
      urlList: [Array],
      body: [Object]
    },
    [Symbol(headers)]: HeadersList {
      cookies: null,
      [Symbol(headers map)]: [Map],
      [Symbol(headers map sorted)]: null
    }
  }
}
```

### What do you see instead?

```
trying
{
  res: Response (1.16 KB) {
    ok: false,
    url: "https://mas.to/api/v1/apps",
    headers: Headers {
      "date": "Mon, 21 Aug 2023 08:50:21 GMT",
      "content-type": "*/*; charset=utf-8",
      "transfer-encoding": "chunked",
      "connection": "keep-alive",
      "x-frame-options": "DENY",
      "x-content-type-options": "nosniff",
      "x-xss-protection": "0",
      "referrer-policy": "same-origin",
      "vary": "Accept, Origin",
      "cache-control": "no-cache",
      "content-security-policy": "base-uri 'none'; default-src 'none'; frame-ancestors 'none'; font-src 'self' https://mas.to; img-src 'self' https: data: blob: https://mas.to; style-src 'self' https://mas.to 'nonce-tAceqs5o41rqOl2cQGCp6Q=='; media-src 'self' https: data: https://mas.to; frame-src 'self' https:; manifest-src 'self' https://mas.to; form-action 'self'; connect-src 'self' data: blob: https://mas.to https://media.mas.to wss://mas.to; script-src 'self' https://mas.to 'wasm-unsafe-eval'; child-src 'self' blob: https://mas.to; worker-src 'self' blob: https://mas.to",
      "strict-transport-security": "max-age=63072000; includeSubDomains",
      "report-to": "{\"endpoints\":[{\"url\":\"https:\\/\\/a.nel.cloudflare.com\\/report\\/v3?s=F3e4PbuJB7s1zuXL2PCMzgXepm9jrkXxF9tUTNk%2F5WcO7vd2rupfn0FDllLy4d6FDK%2FaMByyfiuyAjn%2Blgcb%2BgBBlU%2B1hirV8Z9hxgftM4YVdsoS6xstM4E%3D\"}],\"group\":\"cf-nel\",\"max_age\":604800}",
      "permissions-policy": "interest-cohort=()",
      "x-request-id": "efbab496-9dea-4d3b-a949-4681c3b7fdd0",
      "x-runtime": "0.003401",
      "cf-cache-status": "DYNAMIC",
      "nel": "{\"success_fraction\":0,\"report_to\":\"cf-nel\",\"max_age\":604800}",
      "server": "cloudflare",
      "cf-ray": "7fa1a0c04a4827a7-SEA",
      "alt-svc": "h3=\":443\"; ma=86400"
    },
    statusText: "Not Found",
    redirected: false,
    bodyUsed: false,
    status: 404,
    Blob (1.16 KB)
  }
}
```

### Additional information

looks like it's not including body in the request for some reason. I know the formdata api was recently added...

**Comment by Jarred-Sumner** [comment]
The example above doesn't reproduce the issue. 

Can you provide a complete reproduction?

Here's what I tried:

```js
const server = Bun.serve({
  port: 0,
  fetch(req) {
    return new Response(req.body, { headers: req.headers });
  },
});

const body = new FormData();
body.append("client_name", "tootbooster");
body.append("redirect_uris", `example/123`);
body.append("scopes", "read write push");
const url = `http://${server.hostname}:${server.port}/api/v1/apps`;
const res = await fetch(url, {
  method: "POST",
  body,
});
console.log(await res.formData());
console.log(body);
server.stop(true);
```

**Comment by codegod100** [comment]
In the title it says it's contexed to sveltekit. I know sveltekit interop is a goal for you all so I decided to post here since bun not working with sveltekit isn't really a sveltekit bug

**Comment by codegod100** [comment]
here is code in context of sveltekit project for reference: https://github.com/codegod100/tootbooster/blob/main/src/routes/create/%5Bhost%5D/%2Bpage.server.js#L32

**Comment by codegod100** [comment]
Created a repro repo https://github.com/codegod100/bun-repo

fetch fails on page load with `bunx --bun vite dev`

**Comment by codegod100** [comment]
This might be relevant: https://kit.svelte.dev/docs/load#making-fetch-requests 

I tested using browser fetch and it worked so this is probably not a big issue

**Comment by vyconm** [comment]
we are running into this issue when trying to deploy to an EC2 AWS instance. Latest Bun 1.1.10, probably related to CSP as the console throws an 403-forbidden, and SvelteKit action-forms work again when forcing SvelteKit to ignore CSP - bun doesn't seem to understand/respect the ORIGIN env variable. Might also be somewhat related to node:https in addition, and is our only problem keeping us from deploying our app in Production with bun.

Do you need a newer reproduction for this to be investigated?

**Comment by Jarred-Sumner** [comment]
I'm pretty sure this was fixed in Bun v1.1.25 or so when we updated how fetch() handles arguments parsing and extended `Request` objects.

If you're still running into this, please leave a comment
